src/ui/setup/prefs.ts

```diff
@@ -25,13 +27,55 @@ export interface ReplaySessions {
 export interface ReplaySession {
   viewMode: ViewMode;
   toolboxLayout: ToolboxLayout;
-  selectedPrimaryPanel: PrimaryPanelName;
+  selectedPrimaryPanel: PrimaryPanelName | null;
   selectedPanel: SecondaryPanelName;
   localNags: LocalNag[];
   tabs: Tab[];
   persistedSelectedLocation: PartialLocation | null;
 }
 
+function initializeAsyncStore() {
+  const asyncStore = asyncStoreHelper("devtools", {
+    replaySessions: ["Json", "replay-sessions", {} as Record<string, ReplaySession>],
+  });
+
+  let cachedValue: ReplaySessions | null = null;
+
+  async function readAsyncStore(): Promise<ReplaySessions> {
+    if (cachedValue !== null) {
+      return cachedValue;
+    }
+
+    cachedValue = await asyncStore.replaySessions;
+
+    return cachedValue;
+  }
+
+  function writeAsyncStore(value: ReplaySessions): void {
+    cachedValue = value;
+
+    writeAsyncStoreDebounced(value);
+  }
+
+  const writeAsyncStoreDebounced = debounce((value: ReplaySessions) => {
+    asyncStore.replaySessions = value;
+  }, 250);
+
+  return {
+    readAsyncStore,
+    writeAsyncStore,
+  };
+}
+
+const { readAsyncStore, writeAsyncStore } = initializeAsyncStore();
```

Comment:
This ensures a single path to read/write the store, which makes sure that our cached "current" value is always correct. (Previously it was stale after the first update.)